lib/faktory/testing.rb

```diff
@@ -63,13 +65,16 @@ def self.constantize(str)
       names.shift if names.empty? || names.first.empty?
 
       names.inject(Object) do |constant, name|
-        constant.const_defined?(name) ? constant.const_get(name) : constant.const_missing(name)
+        # the false flag limits search for name to under the constant namespace
+        # which mimics Rails' behaviour
+        constant.const_defined?(name, false) ? constant.const_get(name, false) : constant.const_missing(name)
       end
     end
   end
 
-  # Fake it 'til you make it by default
-  Faktory::Testing.fake!
+  # Test modes have to be opted into explicitly.
+  # Just requiring the testing module shouldn't automatically change the testing mode.
+  Faktory::Testing.disable!
```

Comment:
I figured this would ultimately be the solution to our testing issues.  We'll have to make sure this gets noted in the wiki for testing.  Thanks! :+1: